acceptance/suites/tests/555-puppetserver-service/reboot.rb

```diff
@@ -0,0 +1,55 @@
+if (options[:type] == 'pe') then
+  pssn='pe-puppetserver'
+else
+  pssn='puppetserver'
+end
+
+test_name "Validates the ability of the #{pssn} service to reliably start after a reboot."
+
+max=3
+sleeptime=45
+retryCounter=0
+step "\n-----------\nValidate #{pssn} is up and running, then reboot, repeat #{max} times\n-----------"
+
+for i in 1..max 
+  step "\n-----\nLoop #{i}\n-----"
+   
+  step "check uptime"
+  origUptime=on(master, "awk '{print $1}' /proc/uptime").stdout.chomp.to_i
+
+  step "reboot"
+  #rebooting with on(master,'reboot') causes an IOError: stream end.  This seems like a sane workaround.
+  #QENG-1914 for more...
+  system "ssh -i #{options[:keyfile]} -o StrictHostKeyChecking=no root@#{master.reachable_name} \'reboot\'"
```

Comment:
I'm concerned that using a different SSH client than beaker normally uses will cause the test to depend on the platform specific behavior relative to where it runs.  For example, if I run this beaker test on my laptop I'm subject to the Mac OS X behavior of `ssh`, whereas when it runs in CI it will run based on the GNU Linux behavior of SSH.

Would it be possible to catch the exception thrown when logging in and rebooting the box using beaker's normal transport?